src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+		return DecoderConstant.HTTPS
+				+ getHost()
+				+ path;
+	}
+
+	/**
+	 * This method is used to retrieve Authentication Cookie (SessionID) by send POST request to http://{IP_Address}/apis/authentication
+	 *
+	 * When there is no Session ID data or having an Exception, The Session ID of Authentication Cookie is going to set with null value
+	 */
+	private void retrieveSessionFromDecoder() {
+		String login = getLogin();
+		String password = getPassword();
+
+		ObjectNode request = JsonNodeFactory.instance.objectNode();
+		request.put(DecoderConstant.USERNAME, login);
+		request.put(DecoderConstant.PASSWORD, password);
+
+		try {
+			if (this.authenticationCookie.getSessionID() == null) {
+				HttpClient httpClient = this.obtainHttpClient(true);
+
+				HttpPost httppost = new HttpPost(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.AUTHENTICATION));
+				httppost.setHeader("Content-Type", "text/xml");
+				httppost.setHeader("Content-Type", "application/json");
+				StringEntity entity = new StringEntity(request.toString());
+				httppost.setEntity(entity);
+				HttpResponse response = null;
+
+				// Get SessionID
+				try {
+					response = httpClient.execute(httppost);
+				} finally {
+					if (response instanceof CloseableHttpResponse) {
+						((CloseableHttpResponse) response).close();
+					}
+				}
+				Header headerResponse = response.getFirstHeader(DecoderConstant.SESSION_ID);
+
+				if (headerResponse.getValue() != null) {
+					authenticationCookie.setSessionID(headerResponse.getValue());
+				} else {
+					throw new ResourceNotReachableException(DecoderConstant.GETTING_SESSION_ID_ERR);
+				}
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(DecoderConstant.GETTING_SESSION_ID_ERR);
+		}
+	}
+
+	/**
+	 * This method is used to retrieve User Role by send GET request to http://{IP_Address}/apis/accounts/{username}
+	 *
+	 * @throws ResourceNotReachableException When there is no valid User Role data or having an Exception
+	 */
+	private void retrieveUserRoleBasedFromDecoder() {
+		String login = getLogin();
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				JsonNode response = doGet(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.ROLE_BASED + DecoderConstant.SLASH + login), JsonNode.class);
+				if (response != null) {
+					objectMapper = new ObjectMapper();
+					authenticationInfo = objectMapper.readValue(response.toString(), AuthenticationInfo.class);
+					authenticationInfo.getAuthenticationRole().getRole();
+
+					if (authenticationInfo.getAuthenticationRole().getRole() == null) {
+						throw new ResourceNotReachableException(DecoderConstant.ROLE_BASED_ERR);
+					}
+				} else {
+					throw new ResourceNotReachableException(DecoderConstant.ROLE_BASED_ERR);
+				}
+			} else {
+				throw new ResourceNotReachableException(DecoderConstant.ROLE_BASED_ERR);
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(DecoderConstant.ROLE_BASED_ERR);
```

Comment:
please add the Throwable in this. It should be 
throw new ResourceNotReachableException(DecoderConstant.ROLE_BASED_ERR, e);

Reply:
I fixed it